src/Elastic.Apm/Model/ExecutionSegmentCommon.cs

```diff
@@ -274,5 +274,42 @@ IReadOnlyCollection<string> applicationNamespaces
 				_ => null
 			};
 		}
+
+		/// <summary>
+		/// Captures an error based on a log
+		/// </summary>
+		/// <param name="logOnError"></param>
+		/// <param name="payloadSender"></param>
+		/// <param name="logger"></param>
+		/// <param name="executionSegment"></param>
+		/// <param name="configSnapshot"></param>
+		/// <param name="enclosingTransaction"></param>
+		/// <param name="parentId"></param>
+		/// <param name="exception"></param>
+		internal static void CaptureLogAsError(ErrorLog logOnError, IPayloadSender payloadSender, IApmLogger logger,
+			IExecutionSegment executionSegment, IConfigSnapshot configSnapshot, Transaction enclosingTransaction, string parentId,
+			Exception exception = null
+		)
+		{
+			var error = new Error(logOnError, enclosingTransaction, parentId ?? executionSegment?.Id, logger)
+			{
+				Culprit = $"{logOnError.Level ?? "Error"} log"
+			};
+
+			if (exception != null)
+			{
+				error.Exception = new CapturedException { Message = exception.Message, Type = exception.GetType().FullName };
+
+				if (exception.StackTrace != null)
+				{
+					//TODO:
```

Comment:
still TODO or can be removed?

Reply:
Ah - yeah this was still ToDo - I implemented it.